components/eventing-controller/pkg/handlers/utils.go

```diff
@@ -0,0 +1,92 @@
+package handlers
+
+import (
+	"fmt"
+	"strings"
+
+	eventingv1alpha1 "github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha1"
+
+	"github.com/mitchellh/hashstructure"
+
+	types2 "github.com/kyma-project/kyma/components/eventing-controller/pkg/ems2/api/events/types"
+)
+
+func getHash(subscription *types2.Subscription) (int64, error) {
+	if hash, err := hashstructure.Hash(subscription, nil); err != nil {
+		return 0, err
+	} else {
+		return int64(hash), nil
+	}
+}
+
+func getInternalView4Ev2(subscription *eventingv1alpha1.Subscription) (*types2.Subscription, error) {
+	emsSubscription := &types2.Subscription{}
+
+	// Name
+	emsSubscription.Name = subscription.Name
+	emsSubscription.ContentMode = subscription.Spec.ProtocolSettings.ContentMode
+	emsSubscription.ExemptHandshake = subscription.Spec.ProtocolSettings.ExemptHandshake
+
+	// Qos
+	qos := strings.ReplaceAll(subscription.Spec.ProtocolSettings.Qos, "-", "_")
+	if qos == string(types2.QosAtLeastOnce) {
```

Comment:
```suggestion
	if qos == types2.QosAtLeastOnce {
```

Reply:
solves the compile error:
"cannot compare qos == types2.QosAtLeastOnce (mismatched types string and types.Qos)"